KAFKA-2394: move to RollingFileAppender by default for log4j

cotedm opened the pull request.

Commit:
update log4j.properties to use RollingFileAppender

cristim commented on config/log4j.properties:
```diff
@@ -13,47 +13,59 @@
 
-log4j.appender.kafkaAppender=org.apache.log4j.DailyRollingFileAppender
-log4j.appender.kafkaAppender.DatePattern='.'yyyy-MM-dd-HH
+log4j.appender.kafkaAppender=org.apache.log4j.log4j.RollingFileAppender
```
I think the "log4j.log4j" duplication in the kafkaAppender name is incorrect.(also repeated below)

I tried this on one of my machines and I had to de-duplicate that in order to get it work.

cotedm replied:
Yes, that's a typo.  I had tested the config, but must've made a typo when copying over.  Thanks for catching it, latest commit takes care of that.

cristim commented on config/log4j.properties:
```diff
@@ -13,47 +13,59 @@
 log4j.appender.kafkaAppender.layout.ConversionPattern=[%d] %p %m (%c)%n
+log4j.appender.kafkaAppender.Append=true
+log4j.appender.kafkaAppender.MaxBackupIndex=10
+log4j.appender.kafkaAppender.MaxFileSize=100MB
```
I think 10*100MB may be too much, considering that log4j doesn't compress the rotated files.

How about somewhere around 500MB in total per log file?(100 \* 5 or 50 \* 10, it doesn't matter much anyway how it's split) The logs should be sent outside the node in near-real-time on production environments.

cotedm replied:
I think 1GB is an appropriate size for service logging.  500MB is a little bit light when you consider how chatty state-change and controller logs can get when a system is not behaving well.  As you mentioned, logs should be sent outside the node, so saving 500MB of space is probably not a high priority.  10/100MB files is also standard in the hadoop ecosystem where I've found this to work quite well.

cristim replied:
We run Kafka on Docker containers with 12GB of allocated disk, including the OS, and since we have multiple logs and there's no compression(that would be a nice improvement), that may add up to filling up the disk. I know that's quite small, and that disk is cheap nowadays, but it also matters if someone is actually looking at those logs on the machines, otherwise they are useless.

The question is if this config is supposed to be production-ready as it is, or just an example that people are supposed to adjust to their production installations. I for one would prefer to have sane defaults which I don't need to override, but I don't mind much if I need to change these values for our production environment.

cotedm replied:
With Kafka on Docker, you should really be logging to standard out instead of putting your logs on the disk allocated to docker because if the docker container dies, you won't be able to find out what Kafka was doing.  I think the defaults should be appropriate for a small production installation.  In this case, a 12GB disk would seem to be in the "special case" rather than common path when you consider even developers running on laptops will have closer to 100GB of local disk to work with in most cases.  What do you think @cristim ?

cristim replied:
Unfortunately with the current way we run Docker containers we must first write the logs to the local filesystem but we also run a log shipping software inside each container that exports them in near-real-time, so that's not a problem. For now I'll need to decrease the number of rotated logs anyway, that's not a problem.

cotedm replied:
Ah ok, sounds good

cristim reviewed (changes requested):
Doesn't work for me on kafka 0.9.0.1 because of the log4j.log4j duplication.

Was this ever tested on a running installation or has this recently changed in Kafka 0.10.x?

Commit:
remove duplicated log4j.

Commit:
Removing upgrade.html changes 

I'm removing changes to upgrade.html because this isn't really a huge change and probably doesn't need to be noted in the release notes.  It's incompatible, but I don't imagine a user would be looking in the docs for why the change happened

cotedm commented:
retargeting to 0.11, @ewencp you still think this is a good time to change this?

Commit:
Merge branch 'trunk' into KAFKA-2394

cotedm commented:
@hachikuji this would be nice to have in 1.0 I think since it is a more mature log rolling policy. What do you think?

hachikuji commented:
@cotedm Thanks for the patch. I discussed this briefly with @ijuma. We'll likely be upgrading the logging system in the next release since log4j 1.x has been deprecated. Do you think it is better to get this in now or wait until then? Since there are some compatibility implications for both changes, we were thinking it might be better to do them both at once.

cotedm commented:
@hachikuji I believe the log4j 2 configuration paradigm is totally different from the 1.x line (XML vs. key/value pairs) so yes this patch would be effectively a no-op if the plan is to go to log4j 2. In either case, as long as we are cleaning up the logs after some specified time or space limit by default, I'm good with it.
